.github/workflows/test.yaml

```diff
@@ -0,0 +1,36 @@
+name: lambda-test
+
+on:
+  # This is a dispatched workflow to be called as a job in other workflows
+  # https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow
+  workflow_call:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - uses: pre-commit/action@v3.0.0
+
+  pytest:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - run: pip install -U pipenv
+      - run: pipenv install --dev
+      - run: pipenv run python3 -m pytest tests/ -vv
+
+  sam-build-and-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/sam-build
+      - run: sam validate --lint
```

Comment:
i'm wondering if it's required to run `sam-build` before a `sam validate`?  do you know whether `sam validate` checks both the user generated and sam generated template? or only the user generated one?

Reply:
oh wow, I had been assuming that `sam validate` was checking the sam-generated template by default because the end of the output from `sam build` suggests running `sam validate` next. But now that I'm verifying, it looks like the default is to verify the user template, not the sam-generated template.